staging/src/k8s.io/client-go/discovery/cached_discovery.go

```diff
@@ -238,8 +238,32 @@ func (d *CachedDiscoveryClient) Invalidate() {
 	d.invalidated = true
 }
 
+// NewCachedDiscoveryClientForConfig creates a new DiscoveryClient for the given config, and wraps
+// the created client in a CachedDiscoveryClient. The provided configuration is upddated with a
+// custom transport that understands cache responses.
+func NewCachedDiscoveryClientForConfig(config *restclient.Config, discoveryCacheDir, httpCacheDir string, ttl time.Duration) (*CachedDiscoveryClient, error) {
```

Comment:
Document why we have two cache dirs, even if it is just historical with a TODO to figure out how to stop having two.

Reply:
and doc behavior on empty